src/main/scala/com/gu/formstack/config/FormstackGuardianMetadata.scala

```diff
@@ -0,0 +1,41 @@
+package com.gu.formstack.config
+
+import com.gu.formstack.Logging
+import com.gu.formstack.services.FormstackClient.Form
+
+import java.time.temporal.TemporalAmount
+import java.time.{ Period, ZoneId, ZonedDateTime }
+import scala.util.Try
+
+class FormstackGuardianMetadata extends Logging {
+
+  // Some forms are used as templates to create other forms.
+  private def isFormATemplate(form: Form): Boolean = {
+    form.name.toLowerCase.contains("template")
+  }
+
+  // Some forms want to be kept for a long time to e.g. gather data over many years.
+  private def isFormLongstanding(form: Form): Boolean = {
+    form.name.contains("_RP")
+  }
+
+  // i.e. should we delete the form, or form submissions if they have expired?
+  def isFormGDPREligible(form: Form): Boolean = {
+    !isFormATemplate(form) && !isFormLongstanding(form)
+  }
+
+  val submissionExpiryPeriod: TemporalAmount = Period.ofDays(390)
+
+  def isFormExpired(form: Form): Try[Boolean] = {
+    for {
+      zoneId <- Try(ZoneId.of(form.timezone))
+      localExpiryTime = form.lastSubmissionTime
+        .map(_.plusDays(390))
+        .orElse(form.updated.map(_.plusDays(30)))
+        .getOrElse(form.created.plusDays(30))
+      zonedExpiryTime = localExpiryTime.atZone(zoneId)
```

Comment:
Are `.updated` and `.created` assessed from the perspective of housekeeping for forms that haven't been edited in 30 days? I'm assuming this is directly lifted from the sample scripts provided but I wonder if it might ever take more than 30 days from creation / final edit to the first user submissions coming in (given that form creation is a relatively easy process compared to integration into the site)

Reply:
`.updated` and `.created` are assessed if a form has never received a submission - I'll add a comment to this effect in the code. 

Yeah, this was lifted from the previous scripts. I guess that given these were previously running, 30 days is ok; but as CP are running these rules by various journalists, we'll get confirmation either way.